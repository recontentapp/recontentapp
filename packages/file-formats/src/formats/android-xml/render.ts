import { XMLBuilder } from 'fast-xml-parser'
import { Renderer } from '../../types'
import { ATTRIBUTE_NAME_PREFIX, TEXT_NODE_NAME } from './constants'

export const renderAndroidXML: Renderer = data => {
  const builder = new XMLBuilder({
    format: true,
    ignoreAttributes: false,
    textNodeName: TEXT_NODE_NAME,
    attributeNamePrefix: ATTRIBUTE_NAME_PREFIX,
  })

  const formattedData = {
    '?xml': {
      '@@version': '1.0',
      '@@encoding': 'utf-8',
    },
    resources: {
      string: Object.entries(data).map(([key, value]) => ({
        '@@name': key,
        [TEXT_NODE_NAME]: value,
      })),
    },
  }

  const result = builder.build(formattedData)

  return Promise.resolve(Buffer.from(result))
}
