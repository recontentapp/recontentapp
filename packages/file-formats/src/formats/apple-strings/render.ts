import { Renderer } from '../../types'

const escapeString = (str: string) => {
  return str.replace(/\"/gim, '\\"')
}

export const renderAppleStrings: Renderer = data => {
  const result = Object.entries(data).map(([key, value]) => {
    return `"${key}" = "${escapeString(value)}";`
  })

  return Promise.resolve(Buffer.from(result.join('\n')))
}
