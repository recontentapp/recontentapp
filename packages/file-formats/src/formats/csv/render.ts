import { Renderer } from '../../types'
import { DELIMITER } from './constants'

export const renderCSV: Renderer = data => {
  const csv = Object.entries(data).reduce<string>((result, [key, value]) => {
    return `${result}${key}${DELIMITER}${value}\n`
  }, '')

  return Promise.resolve(Buffer.from(csv))
}
