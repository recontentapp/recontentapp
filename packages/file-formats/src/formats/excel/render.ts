import * as Excel from 'exceljs'
import { Renderer } from '../../types'
import { escape } from './functions'

export const renderExcel: Renderer = async data => {
  const workbook = new Excel.Workbook()
  const sheet = workbook.addWorksheet('Translations')

  sheet.addRow(['Key', 'Translation'])

  Object.entries(data).forEach(([key, translation]) => {
    sheet.addRow([escape(key), escape(translation)])
  })

  const buffer = await workbook.xlsx.writeBuffer()
  return buffer as Buffer
}
