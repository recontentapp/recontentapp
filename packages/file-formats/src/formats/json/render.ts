import { Renderer } from '../../types'

export const renderJSON: Renderer = data => {
  return Promise.resolve(Buffer.from(JSON.stringify(data, null, 2)))
}

export const renderNestedJSON: Renderer = data => {
  const obj = Object.keys(data).reduce((result, key) => {
    key.split('.').reduce((res, k, i, arr) => {
      // @ts-expect-error TODO
      return res[k] || (res[k] = arr.length - 1 === i ? data[key] : {})
    }, result)

    return result
  }, {})

  return Promise.resolve(Buffer.from(JSON.stringify(obj, null, 2)))
}
