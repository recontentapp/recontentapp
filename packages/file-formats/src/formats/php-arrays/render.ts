import { Renderer } from '../../types'

export const renderPHPArrays: Renderer = data => {
  const result: string[] = ['<?php', '']

  for (const key in data) {
    result.push(`$lang["${key}"] = "${data[key]}";`)
  }

  result.push('')

  return Promise.resolve(Buffer.from(result.join('\n')))
}
