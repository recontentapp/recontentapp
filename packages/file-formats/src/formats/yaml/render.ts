import { stringify as yamlStringify } from 'yaml'
import { Renderer } from '../../types'

export const renderYAML: Renderer = data => {
  return Promise.resolve(Buffer.from(yamlStringify(data)))
}

export const renderNestedYAML: Renderer = data => {
  const obj = Object.keys(data).reduce((result, key) => {
    key.split('.').reduce((res, k, i, arr) => {
      // @ts-expect-error TODO
      return res[k] || (res[k] = arr.length - 1 === i ? data[key] : {})
    }, result)
    return result
  }, {})

  return Promise.resolve(Buffer.from(yamlStringify(obj)))
}
