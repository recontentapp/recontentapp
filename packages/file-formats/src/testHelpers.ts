import { Dict } from './types'

export const testDict: Dict = {
  title: 'Hello World',
  'dashboard.title': 'Welcome to your dashboard',
  'dashboard.description':
    'Here you can find all the information you need to get started',
}
