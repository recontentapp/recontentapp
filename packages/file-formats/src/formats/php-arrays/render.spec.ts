import { testDict } from '../../testHelpers'
import { renderPHPArrays } from './render'

describe('phpArrays', () => {
  it('matches snapshot', async () => {
    const result = await renderPHPArrays(testDict)
    expect(result.toString()).toMatchSnapshot()
  })
})
