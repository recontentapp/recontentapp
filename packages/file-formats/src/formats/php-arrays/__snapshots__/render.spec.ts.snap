// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`phpArrays matches snapshot 1`] = `
"<?php

$lang["title"] = "Hello World";
$lang["dashboard.title"] = "Welcome to your dashboard";
$lang["dashboard.description"] = "Here you can find all the information you need to get started";
"
`;
